### ISSUE TITLE
Players Falling Through Transport on Change Map

### ISSUE THREAD

**Issue by cala** [cmangos/issues#198]
This issue is classic-only

_Revisions_
**Core:**  cmangos-classic revision z2336 (fresh install)
    First appearance: was always there
**DB:** classicDB v1.1 rev 0097 (was present before)
**SD2:** s2666 (not related to SD2: same behavior with or without SD2)

_How to reproduce the issue_
- Step onto a boat or zeppelin going from Eastern Kingdoms to Kalimdor (or vice versa)
- Wait for the loading of the target continent map

_What happens_
- Player falls through transport to water (all boats, some zeppelins) or ground (some zeppelins) and dies (fatigue in water, fall damage on ground)
- Transport goes on its way like nothing happened
- Happens 100% of the time with Mac Client
- Happens sometimes with Win Client (I don't know the percentage as I do not have such a client)

_What should happens_
- Players stays on transports and arrives to its destination _alive_.

_Suggestions_
The issue seems to happen right after the player is teleported from the source map to the destination map. Fixable as I tested it on some private server. They claimed to have fix the issue by moving player slightly upward along the Z-axis after teleport and by simulating a "press forward key" by the client to make the player stick to the transport.

Was previously discussed http://getmangos.com/bb/topic/420/zero-help-on-fixing-player-falling-from-transports/ with some suggested fix from ProjectSilverpine.

Note: this issue occurred for a short time back in Vanilla official.

**Comment by MarcoPolo87** [comment]
As posted in the referenced issue:

"Can you please elaborate further. After 5 hours of testing, and riding the three boats & three zeppelins (that swap continents), I was unable to reproduce the issue.of falling off any of the transports.

As for the Feathermoon Ferry...
1) It works fine from Feathermoon Dock.
2) It does not show up when on you are standing on the Dock on the Feralas Mainland.
3) Speculation is that the ship paths outside the "loaded zone" and that would be where it may get "stuck."

The only "transport" that took a long time (that was working as intended) was the Grom'Gol zeppelin from Brill. This zeppelin may also suffer from the same issue as the Feathermoon Dock. It doesn't seem to reliably show up (as in time between arrivals varies).

Also, please list the Classic DB version and the Mangos Server Core version you are running. It helps the Developers to know which version you are running.

Classic DB version 1.5.0 'The Great Swimming Pool'. For core z2656 and SD2 z2734
ACID 1.12.x (Classic) for CMaNGOS-CLASSIC (1.12.X Client)
required_z2621_s2263_12823_04_mangos_creature"

The server is running on a Windows 2012 machine, and I did the tests with a Windows 8.1 machine.

**Comment by cala** [comment]
The falling through transport issue only occurs to Classic Mac client.

As for the "transport never arrives", it is probably affected by activation distance in settings.

**Comment by MarcoPolo87** [comment]
Can you elaborate on the activation distance, in the settings file?

I did not see anything in the mangosd.conf that specified the activation distance 

Is "Visibility.Distance.Continents" the variable that would increase the activation distance?

**Comment by Saben65** [comment]
Hackfix from ProjectSilverpine https://github.com/ProjectSilverpine/server/commit/dfe8e0daff6192ae115d969e06c7d1a152cb283e that @cala was talking about.

**Comment by NocFenix** [comment]
The ProjectSilverpine code is completely different... Has anyone come up with a fix for the existing code base??

**Comment by AnonXS** [comment]
resolved

**Comment by cala** [comment]
I'd like to know how it was resolved because I still have the issue as of today. Like 100% of the time.

**Comment by AnonXS** [comment]
@cala we reconfirmed in discord and @Phatcat said its not happening anymore.

**Comment by cala** [comment]
It is happening to me right now, three times in a row in a freshly built core.
But from what you are saying, I may need to quote myself:
> The falling through transport issue only occurs to Classic Mac client.

😉

**Comment by AnonXS** [comment]
Ah, okay, well thats another story.

**Comment by cyberium** [comment]
This will not be fixed untill proper Player/npc on transport is implemented.

**Comment by cala** [comment]
> This will not be fixed untill proper Player/npc on transport is implemented.

Why do you make me cry?

**Comment by Daedalus-code** [comment]
I experience this on Ubuntu.

**Comment by yosefl20** [comment]
I'm using a mac client with 1.12. The bug could be reproduced in 100%. Currently I'm trying to make it work around by modifying Transports.cpp. No luck till now :(

**Comment by cala** [comment]
After recents updates regarding transports (pets are indeed boarding on transport now, which is nice), I tested [the latest version of the Classic core](https://github.com/cmangos/mangos-classic/commit/040bc47c1aca142199293d15b1fb6c633a6e4257) against this issue and I'm sad to say that things are getting worse : I can't tell if players are still falling from transport because they are now teleported *below* map and even in a wrong position on the map : 
- when going from Menethil (Wetlands, on map 0) to Auberdine (Darkshore, on map 1), they are teleported below Mulgore map (map 1) which is far from Darkshore (and far from any sea water by the way...)
- when going from Menethil (Wetlands, on map 0) to Theramore (Dustwallow Marsh, on map 1), they are teleported below Feralas map (map 1), which is at a good distance from Theramore

Previously, there was still the possibility for players to swim to the shore after falling from transport, now they are guaranteed a certain death by falling below map. :(

**Comment by killerwife** [comment]
I just tested that this does not indeed happen on classic core on a windows client. Did you by any chance extract the new needed mmtiles for transports using --onlyGO switch? It will likely not help but I have to suggest it anyhow. My guess would be that the transport is being sent too late to the client and for some unknown reason the vanilla client being more susceptible to it. You are being summoned to the transport LOCAL coordinates, and what the client does it resolves to the actual world coordinates appropriately. It just maybe turns out the other clients are able to cope with it better.

**Comment by killerwife** [comment]
@cala 
In void WorldSession::HandlePlayerLogin(LoginQueryHolder* holder)
try removing the SMSG_LOGIN_VERIFY_WORLD transport block. Or check if its being correctly filled and not 0 0 0 0.

**Comment by cala** [comment]
I'm currently rebuilding all mmaps (just in case).
I'll check that once I went through that process. Thanks for the help!

**Comment by cala** [comment]
A few more inputs regarding this (updated) issue:
- the mmtiles are present (and re-extracted) but that did not prevent the issue from happening: player fall through the map after teleporting to the other map
- `transportPosition` is filled with what would seems like local coordinates (they are not null)
- using `.gps` right before and right after the teleport shows that x, y, positions of the player are roughly the same (i.e. it seems that the player is teleported on map 1 with the absolute coordinates it had on map 0). It's hard to tell if it is also true for the z position as the player is falling

If you point me to the function used when teleporting, I can check the exact values in here.

**Comment by killerwife** [comment]
Search for SMSG_NEW_WORLD opcode.

**Comment by cala** [comment]
I've got some values at the time of the teleport, in two different runs (same transport, same destination):

We can see that coordinates barely vary from one run to another (which is expected) and more importantly that last coordinates of the player in his/her original map are used as input parameter (and thus as target) when teleporting to the destination map, resulting in the falling below map due to the low Z value from original map.

```
Run 1 (Auberdine -> Menethil)

Input parameters of Player::TeleportTo()

x:5983.15137
y:1180.13184
z:21.4280434
mapid:0

Player's coordinates when Player::TeleportTo() is called:
PositionX:5983.8164
PositionY:1180.81494
PositionZ:21.4280434
MapId:1

======

Run 2 (Auberdine -> Menethil)

Input parameters of Player::TeleportTo()
x:5982.86962
y:1178.94141
z:21.5882664
mapid:0

Final teleport coordinates in Player::TeleportTo() right before sending packet to client:
final_x:5982.86962
final_y:1178.94141
final_z:21.5882664

Player's coordinates when Player::TeleportTo() is called:
PositionX:5984.13477
PositionY:1179.44861
PositionZ:21.5882664
MapId:1
```

**Comment by killerwife** [comment]
@cala test this : https://github.com/cmangos/mangos-classic/commit/72f3be237e6269fddf5be68d870fdb27f402453e

**Comment by cala** [comment]
I will test this tonight. Thanks.

**Comment by cala** [comment]
After testing: the coordinates teleport issue is fixed: players are correctly teleported to the destination map at the beginning of the transport path, on the transport.

Unfortunately, we are nearly back to the orginal issue. I say nearly because:
- either the player still falls through transport in water (like before)
- either the client crashes and when logging back, the player is on the transport half of the time/falls from it into water the other half

I would say that the client crash occurs like 80% of the time. Which still annoying though less than falling under map.

Below is the crash report I got:
 ```
Process:               World of Warcraft [17591]
Path:                  /Applications/Jeux/*/World of Warcraft.app/Contents/MacOS/World of Warcraft
Identifier:            com.blizzard.worldofwarcraft
Version:               1.12.1 (1.12.1)
Code Type:             X86 (Native)
Parent Process:        ??? [1]
Responsible:           World of Warcraft [17591]
User ID:               502

Date/Time:             2020-12-14 21:38:10.582 +0100
OS Version:            Mac OS X 10.14.6 (18G6042)
Report Version:        12
Anonymous UUID:        C95E6E97-37E6-6ADE-2CA4-E778F62B2BC9

Sleep/Wake UUID:       47895A53-881F-4618-A869-078C37240D85

Time Awake Since Boot: 810000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

VM Regions Near 0:
--> 
    __TEXT                 0000000000001000-0000000000693000 [ 6728K] r-x/rwx SM=COW  /Applications/Jeux/*/World of Warcraft.app/Contents/MacOS/World of Warcraft

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   com.blizzard.worldofwarcraft  	0x0026ca19 0x1000 + 2538009
1   com.blizzard.worldofwarcraft  	0x00225432 0x1000 + 2245682
2   com.blizzard.worldofwarcraft  	0x0055122a 0x1000 + 5571114
3   com.blizzard.worldofwarcraft  	0x0009acf7 0x1000 + 630007
4   com.blizzard.worldofwarcraft  	0x00253264 0x1000 + 2433636
5   com.blizzard.worldofwarcraft  	0x000a0456 0x1000 + 652374
6   com.blizzard.worldofwarcraft  	0x00069de9 0x1000 + 429545
7   com.blizzard.worldofwarcraft  	0x0000cac2 0x1000 + 47810
8   com.blizzard.worldofwarcraft  	0x001d3f26 0x1000 + 1912614
9   com.blizzard.worldofwarcraft  	0x001da420 0x1000 + 1938464
10  com.blizzard.worldofwarcraft  	0x001da6f6 0x1000 + 1939190
11  com.blizzard.worldofwarcraft  	0x0000f1ed 0x1000 + 57837
12  com.blizzard.worldofwarcraft  	0x00004085 0x1000 + 12421
13  com.apple.HIToolbox           	0x92e4f9a8 _InvokeEventHandlerUPP(OpaqueEventHandlerCallRef*, OpaqueEventRef*, void*, long (*)(OpaqueEventHandlerCallRef*, OpaqueEventRef*, void*)) + 26
14  com.apple.HIToolbox           	0x92de8fdf DispatchEventToHandlers(EventTargetRec*, OpaqueEventRef*, HandlerCallRec*) + 1970
15  com.apple.HIToolbox           	0x92de8176 SendEventToEventTargetInternal(OpaqueEventRef*, OpaqueEventTargetRef*, HandlerCallRec*) + 376
16  com.apple.HIToolbox           	0x92de7ff2 SendEventToEventTargetWithOptions + 40
17  com.apple.HIToolbox           	0x92e02f4e ToolboxEventDispatcherHandler(OpaqueEventHandlerCallRef*, OpaqueEventRef*, void*) + 1303
18  com.apple.HIToolbox           	0x92de9459 DispatchEventToHandlers(EventTargetRec*, OpaqueEventRef*, HandlerCallRec*) + 3116
19  com.apple.HIToolbox           	0x92de8176 SendEventToEventTargetInternal(OpaqueEventRef*, OpaqueEventTargetRef*, HandlerCallRec*) + 376
20  com.apple.HIToolbox           	0x92e034f9 SendEventToEventTarget + 33
21  com.apple.HIToolbox           	0x92f83329 ToolboxEventDispatcher + 82
22  com.apple.HIToolbox           	0x92f831eb RunApplicationEventLoop + 192
23  com.blizzard.worldofwarcraft  	0x000041b6 0x1000 + 12726
24  com.blizzard.worldofwarcraft  	0x00003172 0x1000 + 8562
25  com.blizzard.worldofwarcraft  	0x0000308d 0x1000 + 8333

Thread 1:: OGL Profiler
0   libsystem_kernel.dylib        	0xa7d551c2 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0xa7d5570f mach_msg + 47
2   com.apple.opengl              	0x9a2b0b17 glcDebugListener + 416
3   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
4   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
5   libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 2:
0   libsystem_kernel.dylib        	0xa7d587aa __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0xa7e0bb46 _pthread_cond_wait + 738
2   libsystem_pthread.dylib       	0xa7e0b1fa pthread_cond_timedwait + 27
3   com.blizzard.worldofwarcraft  	0x0019f1c5 0x1000 + 1696197
4   com.blizzard.worldofwarcraft  	0x00089131 0x1000 + 557361
5   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
6   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
7   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
8   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
9   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
10  libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
11  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 3:
0   libsystem_kernel.dylib        	0xa7d587aa __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0xa7e0bb46 _pthread_cond_wait + 738
2   libsystem_pthread.dylib       	0xa7e095d5 pthread_cond_wait + 25
3   com.blizzard.worldofwarcraft  	0x0019f2cb 0x1000 + 1696459
4   com.blizzard.worldofwarcraft  	0x0003287d 0x1000 + 202877
5   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
6   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
7   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
8   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
9   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
10  libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
11  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 4:: com.apple.audio.IOThread.client
0   libsystem_kernel.dylib        	0xa7d551c2 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0xa7d5570f mach_msg + 47
2   com.apple.audio.CoreAudio     	0x93460899 HALB_MachPort::SendMessageWithReply(unsigned int, unsigned int, unsigned long, unsigned long, mach_msg_header_t*, bool, unsigned int) + 119
3   com.apple.audio.CoreAudio     	0x9343bd41 HALB_MachPort::SendSimpleMessageWithSimpleReply(unsigned int, unsigned int, int, int&, bool, unsigned int) + 49
4   com.apple.audio.CoreAudio     	0x93438f25 HALC_ProxyIOContext::IOWorkLoop() + 1517
5   com.apple.audio.CoreAudio     	0x9343872c HALC_ProxyIOContext::IOThreadEntry(void*) + 208
6   com.apple.audio.CoreAudio     	0x93468db6 invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned long, unsigned int) + 20
7   com.apple.audio.CoreAudio     	0x93438403 HALB_IOThread::Entry(void*) + 71
8   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
9   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
10  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 5:
0   libsystem_kernel.dylib        	0xa7d587aa __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0xa7e0bb46 _pthread_cond_wait + 738
2   libsystem_pthread.dylib       	0xa7e0b1fa pthread_cond_timedwait + 27
3   com.blizzard.worldofwarcraft  	0x0019f1c5 0x1000 + 1696197
4   com.blizzard.worldofwarcraft  	0x00432c95 0x1000 + 4398229
5   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
6   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
7   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
8   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
9   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
10  libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
11  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 6:
0   libsystem_kernel.dylib        	0xa7d5a3fe __select_nocancel + 10
1   libsystem_kernel.dylib        	0xa7d58375 select + 78
2   com.blizzard.worldofwarcraft  	0x00378a81 0x1000 + 3635841
3   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
4   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
5   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
6   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
7   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
8   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
9   libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 7:
0   libsystem_kernel.dylib        	0xa7d5a3fe __select_nocancel + 10
1   libsystem_kernel.dylib        	0xa7d58375 select + 78
2   com.blizzard.worldofwarcraft  	0x00378a81 0x1000 + 3635841
3   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
4   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
5   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
6   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
7   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
8   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
9   libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 8:
0   libsystem_kernel.dylib        	0xa7d5a3fe __select_nocancel + 10
1   libsystem_kernel.dylib        	0xa7d58375 select + 78
2   com.blizzard.worldofwarcraft  	0x00378a81 0x1000 + 3635841
3   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
4   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
5   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
6   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
7   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
8   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
9   libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 9:
0   libsystem_kernel.dylib        	0xa7d5a3fe __select_nocancel + 10
1   libsystem_kernel.dylib        	0xa7d58375 select + 78
2   com.blizzard.worldofwarcraft  	0x00378a81 0x1000 + 3635841
3   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
4   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
5   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
6   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
7   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
8   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
9   libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 10:
0   libsystem_kernel.dylib        	0xa7d587aa __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0xa7e0bb46 _pthread_cond_wait + 738
2   libsystem_pthread.dylib       	0xa7e0b1fa pthread_cond_timedwait + 27
3   com.blizzard.worldofwarcraft  	0x0019f1c5 0x1000 + 1696197
4   com.blizzard.worldofwarcraft  	0x001cabe6 0x1000 + 1874918
5   com.blizzard.worldofwarcraft  	0x001cac56 0x1000 + 1875030
6   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
7   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
8   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
9   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
10  libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
11  libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
12  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 11:
0   libsystem_kernel.dylib        	0xa7d5a3fe __select_nocancel + 10
1   libsystem_kernel.dylib        	0xa7d58375 select + 78
2   com.blizzard.worldofwarcraft  	0x001ca042 0x1000 + 1871938
3   com.blizzard.worldofwarcraft  	0x001cb00a 0x1000 + 1875978
4   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
5   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
6   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
7   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
8   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
9   libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
10  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 12:: com.apple.NSURLConnectionLoader
0   libsystem_kernel.dylib        	0xa7d551c2 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0xa7d5570f mach_msg + 47
2   com.apple.CoreFoundation      	0x9384e3e7 __CFRunLoopServiceMachPort + 289
3   com.apple.CoreFoundation      	0x9384db11 __CFRunLoopRun + 2896
4   com.apple.CoreFoundation      	0x9384ccc4 CFRunLoopRunSpecific + 584
5   com.apple.CoreFoundation      	0x93865a8d CFRunLoopRunInMode + 82
6   com.apple.CFNetwork           	0x92a8d5f4 -[__CoreSchedulingSetRunnable runForever] + 579
7   com.apple.CFNetwork           	0x92a8d22d +[__CoreSchedulingSetRunnable _run:] + 129
8   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
9   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
10  libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
11  libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
12  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 13:
0   libsystem_kernel.dylib        	0xa7d587aa __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0xa7e0bb46 _pthread_cond_wait + 738
2   libsystem_pthread.dylib       	0xa7e0b1fa pthread_cond_timedwait + 27
3   com.blizzard.worldofwarcraft  	0x0019f1c5 0x1000 + 1696197
4   com.blizzard.worldofwarcraft  	0x0037afe6 0x1000 + 3645414
5   com.blizzard.worldofwarcraft  	0x000b3809 0x1000 + 731145
6   com.blizzard.worldofwarcraft  	0x000b39f6 0x1000 + 731638
7   com.apple.Foundation          	0x951a3392 -[NSThread main] + 39
8   com.apple.Foundation          	0x951a323b __NSThread__start__ + 1220
9   libsystem_pthread.dylib       	0xa7e085f8 _pthread_body + 137
10  libsystem_pthread.dylib       	0xa7e0b7f7 _pthread_start + 78
11  libsystem_pthread.dylib       	0xa7e077ce thread_start + 34

Thread 14:
0   libsystem_pthread.dylib       	0xa7e07788 start_wqthread + 0

Thread 0 crashed with X86 Thread State (32-bit):
  eax: 0x00000001  ebx: 0xbfffec00  ecx: 0x00000000  edx: 0xa8691e74
  edi: 0x00000000  esi: 0xbfffebc4  ebp: 0xbfffebe8  esp: 0xbfffeb90
   ss: 0x00000023  efl: 0x00010282  eip: 0x0026ca19   cs: 0x0000001b
   ds: 0x00000023   es: 0x00000023   fs: 0x00000000   gs: 0x0000000f
  cr2: 0x00000000
  
Logical CPU:     0
Error Code:      0x00000004
Trap Number:     14


```

**Comment by killerwife** [comment]
Jesus. Something is up then. Something is being sent wrongly or in wrong order.

**Comment by cala** [comment]
> Jesus. 

Christmas is coming. Maybe He will help us, then. 😆

**Comment by killerwife** [comment]
@cala pushed more commits. 99% chance they wont resolve anything, but here is to hoping.

**Comment by cala** [comment]
The issue is still the same. However, after doing more tests, I've noticed that when reconnecting after client crash: if transport is moving, player falls from it, if transport is static, player stays on it.

**Comment by killerwife** [comment]
Okay, so, please look into checking all packets being sent to client during the transition on vanilla core. Since this is not reproducible on my local, I will try to help in any way I can. It is also worth noting that perhaps updating the coordinates on logging in might be desirable, but I feel like we already do that. CalculatePassengerPosition is the function that calculates real space coords and that were the issue in the initial teleport. However I am still dumbfounded at the crash. If this were the normal client I would start it in IDA and debug the crash but this way I have no clue what happens. Will welcome @Warlockbugs or @ratkosrb input on how to approach this.

**Comment by cala** [comment]
I am well aware that this is tough for you to debug this without being able to reproduce this and having the issue related to client/server behaviour is not an asset.

I'll do several runs, catching the local and global coordinates of the player before/after map change and see if I can find a pattern you can work on.

**Comment by killerwife** [comment]
Please swing by discord and send me the binary for the client you are using. I will try to look at it through disassembler and perhaps try to at least see which packet crashed it.

**Comment by killerwife** [comment]
@cala I just pushed 2 commits. I discovered part of the protocol being bugged due to some map transfers and multithreading problems. I discovered this when fixing vehicle transport map transfer, and I believe it could fix your crash issue. Do test the last two commits.

**Comment by cala** [comment]
Thank you very much for taking time to look into this. I'm currently unavailable due to the holiday season but I definitely test this when I'm back. And if not, I shall send you the binary file of the Mac 1.12 client.

**Comment by cala** [comment]
After running some tests, there are indeed slight improvements but the whole thing is still buggy on Mac client. Most of the time, the client crashes on map change. I'm sending you the client binary as soon as my Internet issues are fixed (such a lovely beginning for the new year...).

When reconnecting the players after client crash, two options happen:
- if the transport is moving, the player falls from it
- if the transport is stationed, the player sticks on it, but not on the coordinates it had when the client crashed/before map change

These happen like all the time with only two exceptions that I was not able to reproduce twice : 
- on map change, no client crash and the player stays on the transport (but not on the same coordinates it had before map change)
- on map change, no client crash but the transport keeps moving without the player that is frozen in the air (fall resumes as soon as the player starts moving)

While waiting for the client upload, tell me if there is any test that I can do on my side that could help you.

**Comment by jimmybrancaccio** [comment]
I was able to replicate this using `mangos-classic` and the macOS client 1.12.1 on macOS High Sierra 10.13.6.

mangos-classic - `7a5d44d576d18380a8955cdec02ff7dc3fde4e99`
classic-db - `4842700709d4f590acd53cb07b69e82d262a4f0b`

It took about 5 runs between Menethil Harbor and Theramore Isle but it finally happened. I noticed on each go-between my character would move on the boat without me moving the character myself. Finally my character "fell" off the boat and into The Great Sea. My character however didn't die (or hasn't yet, and it's been about 10 minutes since). I did see the fatigue bar which ran out, and now my character is losing but regaining health at a rate which isn't killing her.

I've also noticed the boat come back and my character "gets on" the boat but the boat keeps going and my character is back in the sea.

I saw these log entries in the `mangos` console:

```
Delayed Zone Update: Client sent invalid zoneId for X,Y & MAP Coordinates. GUID: 1 zoneId: 11 Expected 15, Coords: -4291.881348 -5608.145996 11.612689
Delayed Zone Update: Client sent invalid zoneId for X,Y & MAP Coordinates. GUID: 1 zoneId: 15 Expected 11, Coords: -3797.318115 179.868774 12.028242
```

I am not sure if they're related but I figured it would be worth including them just in case.

Please do not hesitate to let me know if there's anything else I can check or do to help resolve this issue!

**Comment by cala** [comment]
Just to say that this still happens as of today and revision https://github.com/cmangos/mangos-classic/commit/df0cfe42872f5c959c8ed079748ec2c8b0f1f429
@jimmybrancaccio description is quite accurate of what happens.

**Comment by killerwife** [comment]
So, there is hope for this issue. Someone needs to add 1.12.x parsing of sniffs to TC WPP, and once that is done we can debug the client server communication through our internal sniffer. I was able to fix many issues on TBC and WOTLK thx to it.
